model.py

```diff
@@ -0,0 +1,209 @@
+        expected_outputs_length.append(len(expected_output))
+
+    # shape: (batch_size)
+    expected_outputs_length = np.array(expected_outputs_length, dtype=np.int32)
+
+    return {
+        step['expected_outputs_length']: expected_outputs_length
+    }
+
+
+def to_numpy_metric_wrapper(metric_fun):
+    def metric(data_inputs, expected_outputs):
+        return metric_fun(np.array(data_inputs)[..., 0], np.array(expected_outputs)[..., 0])
+
+    return metric
+
+
+class SignalPredictionPipeline(Pipeline):
+    BATCH_SIZE = 5
+    LAMBDA_LOSS_AMOUNT = 0.003
+    OUTPUT_DIM = 2
+    INPUT_DIM = 2
+    HIDDEN_DIM = 12
+    LAYERS_STACKED_COUNT = 2
+    LEARNING_RATE = 0.006
+    LR_DECAY = 0.92
+    MOMENTUM = 0.5
+    OUTPUT_SIZE = 5
+    EPOCHS = 150
+
+    def __init__(self):
+        super().__init__([
+            TensorflowV1ModelStep(
+                create_graph=create_graph,
+                create_loss=create_loss,
+                create_optimizer=create_optimizer,
+                create_feed_dict=create_feed_dict
+            ).set_hyperparams(HyperparameterSamples({
+                'batch_size': self.BATCH_SIZE,
+                'lambda_loss_amount': self.LAMBDA_LOSS_AMOUNT,
+                'output_dim': self.OUTPUT_DIM,
+                'output_size': self.OUTPUT_SIZE,
+                'input_dim': self.INPUT_DIM,
+                'hidden_dim': self.HIDDEN_DIM,
+                'layers_stacked_count': self.LAYERS_STACKED_COUNT,
+                'learning_rate': self.LEARNING_RATE,
+                'lr_decay': self.LR_DECAY,
+                'momentum': self.MOMENTUM
+            })),
+        ])
+
+
+if __name__ == '__main__':
+    exercise = 4
+    # We choose which data function to use below, in function of the exericse.
+    if exercise == 1:
+        generate_x_y_data = generate_x_y_data_v1
+    if exercise == 2:
+        generate_x_y_data = generate_x_y_data_v2
+    if exercise == 3:
+        generate_x_y_data = generate_x_y_data_v3
+    if exercise == 4:
+        generate_x_y_data = generate_x_y_data_v4
+
+    pipeline = DeepLearningPipeline(
+        SignalPredictionPipeline(),
+        validation_size=0.15,
+        batch_size=SignalPredictionPipeline.BATCH_SIZE,
+        batch_metrics={'mse': to_numpy_metric_wrapper(mean_squared_error)},
+        shuffle_in_each_epoch_at_train=True,
+        n_epochs=SignalPredictionPipeline.EPOCHS,
+        epochs_metrics={'mse': to_numpy_metric_wrapper(mean_squared_error)},
+        scoring_function=to_numpy_metric_wrapper(mean_squared_error)
+    )
+
+    data_inputs, expected_outputs = generate_x_y_data(isTrain=True, batch_size=SignalPredictionPipeline.BATCH_SIZE)
+    pipeline, outputs = pipeline.fit_transform(data_inputs, expected_outputs)
+
+    mse_train = pipeline.get_epoch_metric_train('mse')
+    mse_validation = pipeline.get_epoch_metric_validation('mse')
+
+    plot_metric(mse_train, mse_validation, xlabel='epoch', ylabel='mse', title='Model Mean Squared Error')
```

Comment:
Could you please upload a preview of that plot / chart in your PR for me to inspect this without running the code locally yet?